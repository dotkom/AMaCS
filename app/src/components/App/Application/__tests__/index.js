import React from 'react';
import { shallow } from 'enzyme';

import { Application } from '../';

describe('Application', () => {
  
  const applicationServiceMock = {
    postApplication: jest.fn()
  };
  
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <Application />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with user info', () => {
    const user = {
      fullname: 'Test User',
      email: 'email@example.com'
    }
    const wrapper = shallow(
      <Application user={user} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('updates user info on new props', () => {
    const user = {
      fullname: 'Test User',
      email: 'email@example.com'
    }
    const wrapper = shallow(
      <Application user={user} />
    );
    wrapper.setProps({
      user: {
        fullname: 'Test2 User',
        email: 'email2@example.com'
      }
    })
    expect(wrapper).toMatchSnapshot();
  });

  it('updates info on onChange from login', () => {
    const user = {
      fullname: 'Test Login',
      email: 'email@example.com'
    }
    const wrapper = shallow(
      <Application user={user} />
    );
    wrapper.find('ConnectedServiceComponent').simulate('change', {
      name: 'Post Login',
      email: 'emaillogin@example.com'
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('updates info on onChange from SelectContainer', () => {
    const user = {
      fullname: 'Test Login',
      email: 'email@example.com'
    }
    const wrapper = shallow(
      <Application user={user} />
    );
    wrapper.find('SelectContainer').simulate('change', [
      "Dotkom"
    ]);
    expect(wrapper).toMatchSnapshot();
  });
});
