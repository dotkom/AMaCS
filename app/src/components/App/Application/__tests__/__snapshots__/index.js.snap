// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application renders correctly 1`] = `
<div
  className="component"
>
  <NavigationButton
    link="/"
  >
    <img
      alt="Pil"
      src="TEST_FILE_STUB"
    />
    Tilbake
  </NavigationButton>
  <div
    className="alternative"
  >
    <h2
      className="header"
    >
      Brukerinfo
    </h2>
    <ConnectedServiceComponent
      info={
        Object {
          "email": "",
          "inputEnabled": false,
          "name": "",
        }
      }
      loggedIn={false}
      onChange={[Function]}
    />
  </div>
  <div
    className="content selectWrapper"
  >
    <h2
      className="header"
    >
      Velg komite(er)
    </h2>
    <SelectContainer
      committees={
        Map {
          "arrkom" => Object {
            "icon": "SVGMockComponent",
            "id": 1,
            "info": "Arrangementskomiteen jobber for at informatikkstudentene skal ha sosiale
            tilbud gjennom hele året. Vi arrangerer velkjente arrangement som
            immatrikuleringsball, blåtur, Åre-tur, kryssfest, filmkvelder og mye
            mer! Liker du at andre skal ha det kult? Er du en sosial og sprudlende
            person selv? Da er kanskje Arrangementskomitéen stedet for deg!",
            "key": "arrkom",
            "name": "Arrkom",
          },
          "bedkom" => Object {
            "icon": "SVGMockComponent",
            "id": 3,
            "info": "Bedriftskomiteen er bindeleddet mellom næringslivet og informatikkstudenter,
            og skal i tillegg samarbeide med andre linjeforeninger på NTNU for å ivareta informatikkstudentenes interesser.
            Som medlem vil du ha flere oppgaver, blant annet, å planlegge og arrangere bedriftspresentasjoner, legge ut annonser på våre nettsider, mm. 
            Gjennom bedriftskomiteen vil du opprette en tidlig og god kontakt med næringslivet
            og potensielle arbeidsgivere under studietiden, samtidig som du får et
            innblikk i hvordan bedrifter rekrutterer studenter.",
            "key": "bedkom",
            "name": "Bedkom",
          },
          "dotkom" => Object {
            "icon": "SVGMockComponent",
            "id": 4,
            "info": "Drifts- og utviklingskomiteen har ansvaret for alle linjeforeningens
            datasystemer. Vi leverer tjenester som nettside, wiki,
            e-postsystemer, filområder og mye annet spennende. I en stor
            linjeforening er informasjonsflyt og kvalitet viktig. Drifts- og
            utviklingskomiteen prøver hele tiden å lage smarte løsninger som
            holder en høy standard, slik som i en bedrift. Er du interessert i
            programmering, drifting eller teknologi generelt? Vi kan tilby et
            unikt verv i en meget sosial gjeng som driver med IT på høyt nivå.
            Ingen forkunnskaper er påkrevd annet enn stor lærevilje!",
            "key": "dotkom",
            "name": "Dotkom",
          },
          "fagkom" => Object {
            "icon": "SVGMockComponent",
            "id": 6,
            "info": "Fag- og kurskomiteen tilbyr kunnskap om ny og spennende teknologi,
            gjennom blant annet kurs og foredrag. Vår jobb er å stimulere
            studentenes faglige interesser. Vi oppsøker kompetansen i
            næringslivet, og samarbeider tett med bedriftskomiteen for å komme i
            kontakt med de aktuelle bedriftene. Som medlem av Fag- og kurskomiteen
            vil du få god innsikt i emner utenfor skolen, og samtidig knytte
            tettere bånd til din fremtidige arbeidsgiver.",
            "key": "fagkom",
            "name": "Fagkom",
          },
          "prokom" => Object {
            "icon": "SVGMockComponent",
            "id": 7,
            "info": "Profil- og aviskomiteen sørger for at Onlines ansikt utad er
            profesjonelt og konsistent. Alt profileringsmateriale som klær,
            annonser, brev og brosjyrer, plakater, medaljer og andre effekter
            lages av oss. Vi har også en redaksjon som er ansvarlig for
            linjeforeningens tidsskrift, Offline. Har du en designerspire eller en
            skribent i magen? Vi leter etter deg som er interessert i design,
            markedsføring, journalisme eller skriving!",
            "key": "prokom",
            "name": "Prokom",
          },
          "trikom" => Object {
            "icon": "SVGMockComponent",
            "id": 8,
            "info": "Trivselskomiteen har ansvaret for trivsel blant informatikkstudentene.
            Dette inkluderer å arrangere flere små arrangementer og hyggelige sammenkomster i løpet av året, og drift av kontoret.
            Vi arrangerer frokoster og serievisninger på kontoret, teltturer, juleverksted og andre ting som skaper trivsel.
            Vi er også ansvarlig for drift av kontoret, som vil si å sette opp kontorvakter, møter, administrere tilganger og passe på at det er tilstrekkelig med mat, drikke, rekvisitter og lignende.
            Så hvis du har en gledesspreder i deg og liker å spre trivsel til dine medstudenter, er trikom komiteen for deg!",
            "key": "trikom",
            "name": "Trikom",
          },
        }
      }
      onChange={[Function]}
      onOrderedChange={[Function]}
      ordered={true}
      prioritized={false}
      selected={Array []}
    />
  </div>
  <div
    className="alternative application"
  >
    <h2
      className="header"
    >
      Søknadstekst
    </h2>
    <p>
      Send inn din søknad hvor du forteller litt om deg selv, din motivasjon for vervet, og eventuelt tidligere erfaring fra lignende arbeid. Hold gjerne søknaden kort ettersom at vi arrangerer et intervju med hver søker så vi kan bli enda bedre kjent med deg.
    </p>
    <TextArea
      onChange={[Function]}
      placeholder="Din søknadstekst..."
      value=""
    />
    <Button
      disabled={false}
      onClick={[Function]}
      text="Send søknad"
    />
  </div>
</div>
`;

exports[`Application renders correctly with user info 1`] = `
<div
  className="component"
>
  <NavigationButton
    link="/"
  >
    <img
      alt="Pil"
      src="TEST_FILE_STUB"
    />
    Tilbake
  </NavigationButton>
  <div
    className="alternative"
  >
    <h2
      className="header"
    >
      Brukerinfo
    </h2>
    <ConnectedServiceComponent
      info={
        Object {
          "email": "email@example.com",
          "inputEnabled": false,
          "name": "Test User",
        }
      }
      loggedIn={true}
      onChange={[Function]}
    />
  </div>
  <div
    className="content selectWrapper"
  >
    <h2
      className="header"
    >
      Velg komite(er)
    </h2>
    <SelectContainer
      committees={
        Map {
          "arrkom" => Object {
            "icon": "SVGMockComponent",
            "id": 1,
            "info": "Arrangementskomiteen jobber for at informatikkstudentene skal ha sosiale
            tilbud gjennom hele året. Vi arrangerer velkjente arrangement som
            immatrikuleringsball, blåtur, Åre-tur, kryssfest, filmkvelder og mye
            mer! Liker du at andre skal ha det kult? Er du en sosial og sprudlende
            person selv? Da er kanskje Arrangementskomitéen stedet for deg!",
            "key": "arrkom",
            "name": "Arrkom",
          },
          "bedkom" => Object {
            "icon": "SVGMockComponent",
            "id": 3,
            "info": "Bedriftskomiteen er bindeleddet mellom næringslivet og informatikkstudenter,
            og skal i tillegg samarbeide med andre linjeforeninger på NTNU for å ivareta informatikkstudentenes interesser.
            Som medlem vil du ha flere oppgaver, blant annet, å planlegge og arrangere bedriftspresentasjoner, legge ut annonser på våre nettsider, mm. 
            Gjennom bedriftskomiteen vil du opprette en tidlig og god kontakt med næringslivet
            og potensielle arbeidsgivere under studietiden, samtidig som du får et
            innblikk i hvordan bedrifter rekrutterer studenter.",
            "key": "bedkom",
            "name": "Bedkom",
          },
          "dotkom" => Object {
            "icon": "SVGMockComponent",
            "id": 4,
            "info": "Drifts- og utviklingskomiteen har ansvaret for alle linjeforeningens
            datasystemer. Vi leverer tjenester som nettside, wiki,
            e-postsystemer, filområder og mye annet spennende. I en stor
            linjeforening er informasjonsflyt og kvalitet viktig. Drifts- og
            utviklingskomiteen prøver hele tiden å lage smarte løsninger som
            holder en høy standard, slik som i en bedrift. Er du interessert i
            programmering, drifting eller teknologi generelt? Vi kan tilby et
            unikt verv i en meget sosial gjeng som driver med IT på høyt nivå.
            Ingen forkunnskaper er påkrevd annet enn stor lærevilje!",
            "key": "dotkom",
            "name": "Dotkom",
          },
          "fagkom" => Object {
            "icon": "SVGMockComponent",
            "id": 6,
            "info": "Fag- og kurskomiteen tilbyr kunnskap om ny og spennende teknologi,
            gjennom blant annet kurs og foredrag. Vår jobb er å stimulere
            studentenes faglige interesser. Vi oppsøker kompetansen i
            næringslivet, og samarbeider tett med bedriftskomiteen for å komme i
            kontakt med de aktuelle bedriftene. Som medlem av Fag- og kurskomiteen
            vil du få god innsikt i emner utenfor skolen, og samtidig knytte
            tettere bånd til din fremtidige arbeidsgiver.",
            "key": "fagkom",
            "name": "Fagkom",
          },
          "prokom" => Object {
            "icon": "SVGMockComponent",
            "id": 7,
            "info": "Profil- og aviskomiteen sørger for at Onlines ansikt utad er
            profesjonelt og konsistent. Alt profileringsmateriale som klær,
            annonser, brev og brosjyrer, plakater, medaljer og andre effekter
            lages av oss. Vi har også en redaksjon som er ansvarlig for
            linjeforeningens tidsskrift, Offline. Har du en designerspire eller en
            skribent i magen? Vi leter etter deg som er interessert i design,
            markedsføring, journalisme eller skriving!",
            "key": "prokom",
            "name": "Prokom",
          },
          "trikom" => Object {
            "icon": "SVGMockComponent",
            "id": 8,
            "info": "Trivselskomiteen har ansvaret for trivsel blant informatikkstudentene.
            Dette inkluderer å arrangere flere små arrangementer og hyggelige sammenkomster i løpet av året, og drift av kontoret.
            Vi arrangerer frokoster og serievisninger på kontoret, teltturer, juleverksted og andre ting som skaper trivsel.
            Vi er også ansvarlig for drift av kontoret, som vil si å sette opp kontorvakter, møter, administrere tilganger og passe på at det er tilstrekkelig med mat, drikke, rekvisitter og lignende.
            Så hvis du har en gledesspreder i deg og liker å spre trivsel til dine medstudenter, er trikom komiteen for deg!",
            "key": "trikom",
            "name": "Trikom",
          },
        }
      }
      onChange={[Function]}
      onOrderedChange={[Function]}
      ordered={true}
      prioritized={false}
      selected={Array []}
    />
  </div>
  <div
    className="alternative application"
  >
    <h2
      className="header"
    >
      Søknadstekst
    </h2>
    <p>
      Send inn din søknad hvor du forteller litt om deg selv, din motivasjon for vervet, og eventuelt tidligere erfaring fra lignende arbeid. Hold gjerne søknaden kort ettersom at vi arrangerer et intervju med hver søker så vi kan bli enda bedre kjent med deg.
    </p>
    <TextArea
      onChange={[Function]}
      placeholder="Din søknadstekst..."
      value=""
    />
    <Button
      disabled={false}
      onClick={[Function]}
      text="Send søknad"
    />
  </div>
</div>
`;

exports[`Application updates info on onChange from SelectContainer 1`] = `
<div
  className="component"
>
  <NavigationButton
    link="/"
  >
    <img
      alt="Pil"
      src="TEST_FILE_STUB"
    />
    Tilbake
  </NavigationButton>
  <div
    className="alternative"
  >
    <h2
      className="header"
    >
      Brukerinfo
    </h2>
    <ConnectedServiceComponent
      info={
        Object {
          "email": "email@example.com",
          "inputEnabled": false,
          "name": "Test Login",
        }
      }
      loggedIn={true}
      onChange={[Function]}
    />
  </div>
  <div
    className="content selectWrapper"
  >
    <h2
      className="header"
    >
      Velg komite(er)
    </h2>
    <SelectContainer
      committees={
        Map {
          "arrkom" => Object {
            "icon": "SVGMockComponent",
            "id": 1,
            "info": "Arrangementskomiteen jobber for at informatikkstudentene skal ha sosiale
            tilbud gjennom hele året. Vi arrangerer velkjente arrangement som
            immatrikuleringsball, blåtur, Åre-tur, kryssfest, filmkvelder og mye
            mer! Liker du at andre skal ha det kult? Er du en sosial og sprudlende
            person selv? Da er kanskje Arrangementskomitéen stedet for deg!",
            "key": "arrkom",
            "name": "Arrkom",
          },
          "bedkom" => Object {
            "icon": "SVGMockComponent",
            "id": 3,
            "info": "Bedriftskomiteen er bindeleddet mellom næringslivet og informatikkstudenter,
            og skal i tillegg samarbeide med andre linjeforeninger på NTNU for å ivareta informatikkstudentenes interesser.
            Som medlem vil du ha flere oppgaver, blant annet, å planlegge og arrangere bedriftspresentasjoner, legge ut annonser på våre nettsider, mm. 
            Gjennom bedriftskomiteen vil du opprette en tidlig og god kontakt med næringslivet
            og potensielle arbeidsgivere under studietiden, samtidig som du får et
            innblikk i hvordan bedrifter rekrutterer studenter.",
            "key": "bedkom",
            "name": "Bedkom",
          },
          "dotkom" => Object {
            "icon": "SVGMockComponent",
            "id": 4,
            "info": "Drifts- og utviklingskomiteen har ansvaret for alle linjeforeningens
            datasystemer. Vi leverer tjenester som nettside, wiki,
            e-postsystemer, filområder og mye annet spennende. I en stor
            linjeforening er informasjonsflyt og kvalitet viktig. Drifts- og
            utviklingskomiteen prøver hele tiden å lage smarte løsninger som
            holder en høy standard, slik som i en bedrift. Er du interessert i
            programmering, drifting eller teknologi generelt? Vi kan tilby et
            unikt verv i en meget sosial gjeng som driver med IT på høyt nivå.
            Ingen forkunnskaper er påkrevd annet enn stor lærevilje!",
            "key": "dotkom",
            "name": "Dotkom",
          },
          "fagkom" => Object {
            "icon": "SVGMockComponent",
            "id": 6,
            "info": "Fag- og kurskomiteen tilbyr kunnskap om ny og spennende teknologi,
            gjennom blant annet kurs og foredrag. Vår jobb er å stimulere
            studentenes faglige interesser. Vi oppsøker kompetansen i
            næringslivet, og samarbeider tett med bedriftskomiteen for å komme i
            kontakt med de aktuelle bedriftene. Som medlem av Fag- og kurskomiteen
            vil du få god innsikt i emner utenfor skolen, og samtidig knytte
            tettere bånd til din fremtidige arbeidsgiver.",
            "key": "fagkom",
            "name": "Fagkom",
          },
          "prokom" => Object {
            "icon": "SVGMockComponent",
            "id": 7,
            "info": "Profil- og aviskomiteen sørger for at Onlines ansikt utad er
            profesjonelt og konsistent. Alt profileringsmateriale som klær,
            annonser, brev og brosjyrer, plakater, medaljer og andre effekter
            lages av oss. Vi har også en redaksjon som er ansvarlig for
            linjeforeningens tidsskrift, Offline. Har du en designerspire eller en
            skribent i magen? Vi leter etter deg som er interessert i design,
            markedsføring, journalisme eller skriving!",
            "key": "prokom",
            "name": "Prokom",
          },
          "trikom" => Object {
            "icon": "SVGMockComponent",
            "id": 8,
            "info": "Trivselskomiteen har ansvaret for trivsel blant informatikkstudentene.
            Dette inkluderer å arrangere flere små arrangementer og hyggelige sammenkomster i løpet av året, og drift av kontoret.
            Vi arrangerer frokoster og serievisninger på kontoret, teltturer, juleverksted og andre ting som skaper trivsel.
            Vi er også ansvarlig for drift av kontoret, som vil si å sette opp kontorvakter, møter, administrere tilganger og passe på at det er tilstrekkelig med mat, drikke, rekvisitter og lignende.
            Så hvis du har en gledesspreder i deg og liker å spre trivsel til dine medstudenter, er trikom komiteen for deg!",
            "key": "trikom",
            "name": "Trikom",
          },
        }
      }
      onChange={[Function]}
      onOrderedChange={[Function]}
      ordered={true}
      prioritized={false}
      selected={
        Array [
          "Dotkom",
        ]
      }
    />
  </div>
  <div
    className="alternative application"
  >
    <h2
      className="header"
    >
      Søknadstekst
    </h2>
    <p>
      Send inn din søknad hvor du forteller litt om deg selv, din motivasjon for vervet, og eventuelt tidligere erfaring fra lignende arbeid. Hold gjerne søknaden kort ettersom at vi arrangerer et intervju med hver søker så vi kan bli enda bedre kjent med deg.
    </p>
    <TextArea
      onChange={[Function]}
      placeholder="Din søknadstekst..."
      value=""
    />
    <Button
      disabled={false}
      onClick={[Function]}
      text="Send søknad"
    />
  </div>
</div>
`;

exports[`Application updates info on onChange from login 1`] = `
<div
  className="component"
>
  <NavigationButton
    link="/"
  >
    <img
      alt="Pil"
      src="TEST_FILE_STUB"
    />
    Tilbake
  </NavigationButton>
  <div
    className="alternative"
  >
    <h2
      className="header"
    >
      Brukerinfo
    </h2>
    <ConnectedServiceComponent
      info={
        Object {
          "email": "emaillogin@example.com",
          "inputEnabled": false,
          "name": "Post Login",
        }
      }
      loggedIn={true}
      onChange={[Function]}
    />
  </div>
  <div
    className="content selectWrapper"
  >
    <h2
      className="header"
    >
      Velg komite(er)
    </h2>
    <SelectContainer
      committees={
        Map {
          "arrkom" => Object {
            "icon": "SVGMockComponent",
            "id": 1,
            "info": "Arrangementskomiteen jobber for at informatikkstudentene skal ha sosiale
            tilbud gjennom hele året. Vi arrangerer velkjente arrangement som
            immatrikuleringsball, blåtur, Åre-tur, kryssfest, filmkvelder og mye
            mer! Liker du at andre skal ha det kult? Er du en sosial og sprudlende
            person selv? Da er kanskje Arrangementskomitéen stedet for deg!",
            "key": "arrkom",
            "name": "Arrkom",
          },
          "bedkom" => Object {
            "icon": "SVGMockComponent",
            "id": 3,
            "info": "Bedriftskomiteen er bindeleddet mellom næringslivet og informatikkstudenter,
            og skal i tillegg samarbeide med andre linjeforeninger på NTNU for å ivareta informatikkstudentenes interesser.
            Som medlem vil du ha flere oppgaver, blant annet, å planlegge og arrangere bedriftspresentasjoner, legge ut annonser på våre nettsider, mm. 
            Gjennom bedriftskomiteen vil du opprette en tidlig og god kontakt med næringslivet
            og potensielle arbeidsgivere under studietiden, samtidig som du får et
            innblikk i hvordan bedrifter rekrutterer studenter.",
            "key": "bedkom",
            "name": "Bedkom",
          },
          "dotkom" => Object {
            "icon": "SVGMockComponent",
            "id": 4,
            "info": "Drifts- og utviklingskomiteen har ansvaret for alle linjeforeningens
            datasystemer. Vi leverer tjenester som nettside, wiki,
            e-postsystemer, filområder og mye annet spennende. I en stor
            linjeforening er informasjonsflyt og kvalitet viktig. Drifts- og
            utviklingskomiteen prøver hele tiden å lage smarte løsninger som
            holder en høy standard, slik som i en bedrift. Er du interessert i
            programmering, drifting eller teknologi generelt? Vi kan tilby et
            unikt verv i en meget sosial gjeng som driver med IT på høyt nivå.
            Ingen forkunnskaper er påkrevd annet enn stor lærevilje!",
            "key": "dotkom",
            "name": "Dotkom",
          },
          "fagkom" => Object {
            "icon": "SVGMockComponent",
            "id": 6,
            "info": "Fag- og kurskomiteen tilbyr kunnskap om ny og spennende teknologi,
            gjennom blant annet kurs og foredrag. Vår jobb er å stimulere
            studentenes faglige interesser. Vi oppsøker kompetansen i
            næringslivet, og samarbeider tett med bedriftskomiteen for å komme i
            kontakt med de aktuelle bedriftene. Som medlem av Fag- og kurskomiteen
            vil du få god innsikt i emner utenfor skolen, og samtidig knytte
            tettere bånd til din fremtidige arbeidsgiver.",
            "key": "fagkom",
            "name": "Fagkom",
          },
          "prokom" => Object {
            "icon": "SVGMockComponent",
            "id": 7,
            "info": "Profil- og aviskomiteen sørger for at Onlines ansikt utad er
            profesjonelt og konsistent. Alt profileringsmateriale som klær,
            annonser, brev og brosjyrer, plakater, medaljer og andre effekter
            lages av oss. Vi har også en redaksjon som er ansvarlig for
            linjeforeningens tidsskrift, Offline. Har du en designerspire eller en
            skribent i magen? Vi leter etter deg som er interessert i design,
            markedsføring, journalisme eller skriving!",
            "key": "prokom",
            "name": "Prokom",
          },
          "trikom" => Object {
            "icon": "SVGMockComponent",
            "id": 8,
            "info": "Trivselskomiteen har ansvaret for trivsel blant informatikkstudentene.
            Dette inkluderer å arrangere flere små arrangementer og hyggelige sammenkomster i løpet av året, og drift av kontoret.
            Vi arrangerer frokoster og serievisninger på kontoret, teltturer, juleverksted og andre ting som skaper trivsel.
            Vi er også ansvarlig for drift av kontoret, som vil si å sette opp kontorvakter, møter, administrere tilganger og passe på at det er tilstrekkelig med mat, drikke, rekvisitter og lignende.
            Så hvis du har en gledesspreder i deg og liker å spre trivsel til dine medstudenter, er trikom komiteen for deg!",
            "key": "trikom",
            "name": "Trikom",
          },
        }
      }
      onChange={[Function]}
      onOrderedChange={[Function]}
      ordered={true}
      prioritized={false}
      selected={Array []}
    />
  </div>
  <div
    className="alternative application"
  >
    <h2
      className="header"
    >
      Søknadstekst
    </h2>
    <p>
      Send inn din søknad hvor du forteller litt om deg selv, din motivasjon for vervet, og eventuelt tidligere erfaring fra lignende arbeid. Hold gjerne søknaden kort ettersom at vi arrangerer et intervju med hver søker så vi kan bli enda bedre kjent med deg.
    </p>
    <TextArea
      onChange={[Function]}
      placeholder="Din søknadstekst..."
      value=""
    />
    <Button
      disabled={false}
      onClick={[Function]}
      text="Send søknad"
    />
  </div>
</div>
`;

exports[`Application updates user info on new props 1`] = `
<div
  className="component"
>
  <NavigationButton
    link="/"
  >
    <img
      alt="Pil"
      src="TEST_FILE_STUB"
    />
    Tilbake
  </NavigationButton>
  <div
    className="alternative"
  >
    <h2
      className="header"
    >
      Brukerinfo
    </h2>
    <ConnectedServiceComponent
      info={
        Object {
          "email": "email2@example.com",
          "inputEnabled": false,
          "name": "Test2 User",
        }
      }
      loggedIn={true}
      onChange={[Function]}
    />
  </div>
  <div
    className="content selectWrapper"
  >
    <h2
      className="header"
    >
      Velg komite(er)
    </h2>
    <SelectContainer
      committees={
        Map {
          "arrkom" => Object {
            "icon": "SVGMockComponent",
            "id": 1,
            "info": "Arrangementskomiteen jobber for at informatikkstudentene skal ha sosiale
            tilbud gjennom hele året. Vi arrangerer velkjente arrangement som
            immatrikuleringsball, blåtur, Åre-tur, kryssfest, filmkvelder og mye
            mer! Liker du at andre skal ha det kult? Er du en sosial og sprudlende
            person selv? Da er kanskje Arrangementskomitéen stedet for deg!",
            "key": "arrkom",
            "name": "Arrkom",
          },
          "bedkom" => Object {
            "icon": "SVGMockComponent",
            "id": 3,
            "info": "Bedriftskomiteen er bindeleddet mellom næringslivet og informatikkstudenter,
            og skal i tillegg samarbeide med andre linjeforeninger på NTNU for å ivareta informatikkstudentenes interesser.
            Som medlem vil du ha flere oppgaver, blant annet, å planlegge og arrangere bedriftspresentasjoner, legge ut annonser på våre nettsider, mm. 
            Gjennom bedriftskomiteen vil du opprette en tidlig og god kontakt med næringslivet
            og potensielle arbeidsgivere under studietiden, samtidig som du får et
            innblikk i hvordan bedrifter rekrutterer studenter.",
            "key": "bedkom",
            "name": "Bedkom",
          },
          "dotkom" => Object {
            "icon": "SVGMockComponent",
            "id": 4,
            "info": "Drifts- og utviklingskomiteen har ansvaret for alle linjeforeningens
            datasystemer. Vi leverer tjenester som nettside, wiki,
            e-postsystemer, filområder og mye annet spennende. I en stor
            linjeforening er informasjonsflyt og kvalitet viktig. Drifts- og
            utviklingskomiteen prøver hele tiden å lage smarte løsninger som
            holder en høy standard, slik som i en bedrift. Er du interessert i
            programmering, drifting eller teknologi generelt? Vi kan tilby et
            unikt verv i en meget sosial gjeng som driver med IT på høyt nivå.
            Ingen forkunnskaper er påkrevd annet enn stor lærevilje!",
            "key": "dotkom",
            "name": "Dotkom",
          },
          "fagkom" => Object {
            "icon": "SVGMockComponent",
            "id": 6,
            "info": "Fag- og kurskomiteen tilbyr kunnskap om ny og spennende teknologi,
            gjennom blant annet kurs og foredrag. Vår jobb er å stimulere
            studentenes faglige interesser. Vi oppsøker kompetansen i
            næringslivet, og samarbeider tett med bedriftskomiteen for å komme i
            kontakt med de aktuelle bedriftene. Som medlem av Fag- og kurskomiteen
            vil du få god innsikt i emner utenfor skolen, og samtidig knytte
            tettere bånd til din fremtidige arbeidsgiver.",
            "key": "fagkom",
            "name": "Fagkom",
          },
          "prokom" => Object {
            "icon": "SVGMockComponent",
            "id": 7,
            "info": "Profil- og aviskomiteen sørger for at Onlines ansikt utad er
            profesjonelt og konsistent. Alt profileringsmateriale som klær,
            annonser, brev og brosjyrer, plakater, medaljer og andre effekter
            lages av oss. Vi har også en redaksjon som er ansvarlig for
            linjeforeningens tidsskrift, Offline. Har du en designerspire eller en
            skribent i magen? Vi leter etter deg som er interessert i design,
            markedsføring, journalisme eller skriving!",
            "key": "prokom",
            "name": "Prokom",
          },
          "trikom" => Object {
            "icon": "SVGMockComponent",
            "id": 8,
            "info": "Trivselskomiteen har ansvaret for trivsel blant informatikkstudentene.
            Dette inkluderer å arrangere flere små arrangementer og hyggelige sammenkomster i løpet av året, og drift av kontoret.
            Vi arrangerer frokoster og serievisninger på kontoret, teltturer, juleverksted og andre ting som skaper trivsel.
            Vi er også ansvarlig for drift av kontoret, som vil si å sette opp kontorvakter, møter, administrere tilganger og passe på at det er tilstrekkelig med mat, drikke, rekvisitter og lignende.
            Så hvis du har en gledesspreder i deg og liker å spre trivsel til dine medstudenter, er trikom komiteen for deg!",
            "key": "trikom",
            "name": "Trikom",
          },
        }
      }
      onChange={[Function]}
      onOrderedChange={[Function]}
      ordered={true}
      prioritized={false}
      selected={Array []}
    />
  </div>
  <div
    className="alternative application"
  >
    <h2
      className="header"
    >
      Søknadstekst
    </h2>
    <p>
      Send inn din søknad hvor du forteller litt om deg selv, din motivasjon for vervet, og eventuelt tidligere erfaring fra lignende arbeid. Hold gjerne søknaden kort ettersom at vi arrangerer et intervju med hver søker så vi kan bli enda bedre kjent med deg.
    </p>
    <TextArea
      onChange={[Function]}
      placeholder="Din søknadstekst..."
      value=""
    />
    <Button
      disabled={false}
      onClick={[Function]}
      text="Send søknad"
    />
  </div>
</div>
`;
